From c76d3cc2f68961b05c6c1a509c24e71ab7473c47 Mon Sep 17 00:00:00 2001
From: mihirlad55 <mihirlad55@gmail.com>
Date: Sun, 26 Jul 2020 03:51:16 +0000
Subject: [PATCH] Update from v1.5.2 to v1.5.3

- Fix a major bug where events would not be raised for any monitors after the
  first monitor if the first monitor did not have a selected client (i.e. was
  viewing an unoccupied tag)
---
 ipc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipc.c b/ipc.c
index 1d435dc..e527e23 100644
--- a/ipc.c
+++ b/ipc.c
@@ -1124,7 +1124,7 @@ ipc_send_events(Monitor *mons, Monitor **lastselmon, Monitor *selmon)
     }
 
     Client *sel = m->sel;
-    if (!sel) return;
+    if (!sel) continue;
     ClientState *o = &m->sel->prevstate;
     ClientState n = {.oldstate = sel->oldstate,
                      .isfixed = sel->isfixed,
-- 
2.27.0

